spock-spring/src/main/java/org/spockframework/spring/xml/MockBeanDefinitionParser.java

```diff
@@ -28,10 +29,18 @@ protected String getBeanClassName(Element element) {
   @Override
   protected void doParse(Element element, BeanDefinitionBuilder builder) {
 
+    if (element.getLocalName().equals("spy") && element.getAttribute("ref").isEmpty()
+          && element.getAttribute("class").isEmpty()) {
+      throw new BeanCreationException(element.getAttribute("id"),
+             "Cannot create spy. Please specify either 'class' or 'ref' attributes.");
+    }
     // configure MockFactory
     builder.addConstructorArgValue(element.getAttribute("class"));
     builder.addPropertyValue("name", element.getAttribute("id"));
     builder.addPropertyValue("mockNature", element.getLocalName());
+    if(!element.getAttribute("ref").isEmpty()) {
```

Comment:
Ref is given priority. I'd love to be able to better specify this in the xsd but can't really. I could be more explicit with startup errors if you prefer and just fail fast, then request the user to specify one or the other but not both.